controllers/managedocs_controller.go

```diff
@@ -528,10 +534,15 @@ func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
 			if err := r.reconcileEgressNetworkPolicy(); err != nil {
 				return ctrl.Result{}, err
 			}
-			if err := r.reconcileIngressNetworkPolicy(); err != nil {
-				return ctrl.Result{}, err
-			}
-			if err := r.reconcileCephIngressNetworkPolicy(); err != nil {
+		}
+		if err := r.reconcileIngressNetworkPolicy(); err != nil {
+			return ctrl.Result{}, err
+		}
+		if err := r.reconcileCephIngressNetworkPolicy(); err != nil {
+			return ctrl.Result{}, err
+		}
+		if r.DeploymentType == providerDeploymentType {
+			if err := r.reconcileProviderAPIServerNetworkPolicy(); err != nil {
```

Comment:
Can we please move the condition into the reconcile function, as a guard at the start of the function? 
This is the pattern we agreed on and we are utilizing it in other reconcile functions. 
You can take a look at `reconcileOnboardingValidationSecret` for an example